lib/mayaHydra/hydraExtensions/sceneIndex/mayaHydraSceneIndex.cpp

```diff
@@ -466,9 +466,13 @@ MayaHydraSceneIndex::MayaHydraSceneIndex(
         _fallbackMaterial = SdfPath::EmptyPath(); // Empty path for hydra fallback material
     });
 
-    // Add our pick handler to the pick handler registry.
-    auto pickHandler = std::make_shared<MayaPickHandler>(*this);
-    TF_AXIOM(MayaHydra::PickHandlerRegistry::Instance().Register(_rprimPath, pickHandler));
+    // Add our pick handler to the pick handler registry if there is none.
+    auto& phr = MayaHydra::PickHandlerRegistry::Instance();
+    _unregisterPickHandler = (phr.RegisteredHandler(_rprimPath) == nullptr);
+    if (_unregisterPickHandler) {
```

Comment:
If we switch renderers on a viewport in multi-viewport mode, the scene index tree is re-built, and we try to register a pick handler again, with the same _rprimPath.  Make sure we register a pick handler only once.